src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9447,7 +9447,7 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
     //if (GetTypeId() == TYPEID_UNIT)
     //    ToCreature()->SetCombatStartPosition(GetPositionX(), GetPositionY(), GetPositionZ());
 
-    if (GetTypeId() == TYPEID_UNIT && !IsPet())
+    if (GetTypeId() == TYPEID_UNIT && !IsPet() && !IsSummon() && IsGuardian())
```

Comment:
does it should be `NOT IsGuardian()` ?  so `!IsGuardian()`?

Reply:
No it's work fine. Look video.